src/Instrumentation/Symfony/src/SymfonyInstrumentation.php

```diff
@@ -80,6 +80,30 @@ public static function register(): void
                 array $params,
                 ?Response $response,
                 ?\Throwable $exception
+            ): void {
+                $scope = Context::storage()->scope();
+                if (null === $scope || null === $exception) {
```

Comment:
So if an exception occurs, this post callback will run, otherwise the `terminate` one will? If so, we should document this.

Reply:
That is correct.
I'm assuming this should be documented in the README.md, right?

Reply:
The examples on [this page](https://symfony.com/doc/current/components/http_kernel.html#http-kernel-working-example) seem to suggest that the `handle` and `terminate` can both run.

So it looks like you'd expect 2 spans to be created and then with these changes only close one of them? I'm not so sure that behaviour is correct.

Reply:
There appears to be a parameter to configure whether `handle` should throw [here](https://github.com/symfony/symfony/blob/7.3/src/Symfony/Component/HttpKernel/HttpKernel.php#L69).

So when `$catch = false` the handle post-hook might have an exception and `terminate` will not have been reached yet. I think this would be fine as it handles the recording of the exception and detaches the scope.

However, if `$catch = false` and there is no exception, then `terminate()` would be called after `handle` and the `terminate` post-hook could try to close a scope which does not belong to it.

Reply:
@ChrisLightfootWild 

1. If `handle` throws an exception which is never caught, then `terminate` will never be reached either, according to the[ `HttpKernelRunner` implementation.](https://github.com/symfony/symfony/blob/7.3/src/Symfony/Component/Runtime/Runner/Symfony/HttpKernelRunner.php)
In this situation, it is correct to close the scope on the `post` hook of `handle`.

2. If `handle` does not throw an exception (or has `$catch = true`), then `terminate`  would run afterwards, closing the scope.

The only problem I see is there is not a 100% guarantee that `terminate` will always be executed after `handle`.

For example, something else might go wrong in `HttpKernelRunner` after `handle` was called, e.g. `fastcgi_finish_request()/litespeed_finish_request()` might crash, `flush()` might crash, etc, leaving the scope still open.

Unsure how much of a real issue this is, in most cases.


Normally this would be solved by hooking into `HttpKernelRunner::run` with both `pre` and `post` hooks, as it covers both `handle` and `terminate`, however `$request` and `$response` are not available as parameters there, so I am at a loss right now.


We need either a compromise or another solution enitrely

Reply:
Thanks for that @dciprian-petrisor - I had another look with less bleary eyes and I finally caught up to you guys 💪